pkg/catalog/types.go

```diff
@@ -72,7 +73,7 @@ type ComponentTypeList struct {
 
 // DevfileComponentTypeList lists all the DevfileComponentType's
 type DevfileComponentTypeList struct {
-	DevfileRegistries map[string]string
+	DevfileRegistries map[string]Registry
```

Comment:
Should we just use a array of Registries here? I couldn't find any operation of fetching with the key.

Reply:
True but does it matter? Also, you might want to retrieve a registry by its name at some point… I left the map that was in the original code for that purpose.

Reply:
It's just that a map might require more memory. But you are right that at some point, we might want to retrieve it by it's name. So let's keep it foe now.